### PR TITLE
network: dhcp6pd: assign addresses in the delegated prefix instead of…

### PR DESCRIPTION
**Pull request by yuwata**
… a subnet prefix /64 on upstream interface

Split out of #21625.

### CODE REVIEW

File: src/network/networkd-dhcp-prefix-delegation.c [hunk 1]
```diff
@@ -570,6 +572,41 @@ static int dhcp6_pd_assign_prefix(
...
+        int r;
+
+        assert(link);
+        assert(link->network);
+        assert(pd_prefix);
+
+        (void) in6_addr_prefix_to_string(pd_prefix, pd_prefix_len, &buf);
+
+        if (link->network->dhcp6_pd_announce)
+                log_link_debug(link, "Ignoring Announce= setting on upstream interface.");
```
Comment (keszybz) [inline src/network/networkd-dhcp-prefix-delegation.c:592]: "upstream" → "uplink" ?

**Review by keszybz — COMMENTED** [review]
Looks reasonable, though I don't grok all the implications.

File: src/network/networkd-dhcp-prefix-delegation.c (hunk 1 above)
Comment (yuwata) [reply src/network/networkd-dhcp-prefix-delegation.c:592]: Well,,, I use `uplink` in the variable name and comments, `"upstream interface"` in the log message or man page.
If you do not like it. I will unify that in another later commit, but I do not think it is necessary though that's not strong opinion.

Comment (keszybz) [reply src/network/networkd-dhcp-prefix-delegation.c:592]: I wasn't aware of the distinct conventions. That's fine.

**Comment by keszybz** [comment]
Unless somebody else wants to review, I guess we should merge this when CI passes.

**Pull request merged**